static/app/views/issueDetails/streamline/sidebar/activityDrawer.tsx

```diff
@@ -20,6 +27,25 @@ interface ActivityDrawerProps {
 }
 
 export function ActivityDrawer({group, project}: ActivityDrawerProps) {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const [filterComments, setFilterComments] = useState(
+    location.query.filter === 'comments'
+  );
+
+  useEffect(() => {
```

Comment:
This doesn't seem right? You reset the filter whenever the location changes?

Reply:
updated to hopefully be more clear - basically this is only being used to set the initial state, after that, we don't need the query param anymore

Reply:
Ah I see! Hmm it seems like the better way to do this would be to use the query param as the source of truth - could you do something like this instead?

```tsx
const [searchParams, setSearchParams] = useSearchParams();
const filter = searchParams.get('filter') ?? 'all'

//...

value={filter}
onChange={(value) => {
  setSearchParams(params => {
    params.set('filter', value);
    return params;
  }, {replace: true})
}}
```

That way when you refresh you'll get the same state